src/main/java/orca/flukes/ndl/ManifestLoader.java

```diff
@@ -227,12 +227,30 @@ private String getPrettyName(Resource r) {
 		return rname;
 	}
 	
-	private void addNodeToInterface(String iface, OrcaNode n) {
-		List<OrcaNode> others = interfaceToNode.get(iface);
+	Set<Resource> sameAs = new HashSet<>();
+	
+	private void addNodeToInterface(Resource iface, OrcaNode n) {
+
+		String ifaceName = getTrueName(iface);
+		GUI.logger().debug("Considering adding interface " + ifaceName + " to the list");
+		
+		Resource sameInt = NdlCommons.getSameAsResource(iface);
+		if (sameInt != null) {
+			GUI.logger().debug("Interface " + iface + " same as " + sameInt);
+			// add to the set of duplicates and remove any previous copies
+			sameAs.add(sameInt);
+			interfaceToNode.remove(getTrueName(sameInt));
```

Comment:
I would be useful to see a unit test for this code.  I'm having trouble seeing how this all works.  Maybe I don't understand how the results (`interfaceToNode`?) are used.

If we have two (duplicate / SameAs) interfaces, A and A', what happens when we run this function with both of those interfaces?

## A
- Function is called with interface `A`
- The SameAs interface `A'` is found, and added to `sameAs`
- `A'` is removed from `interfaceToNode` -- but it isn't in there yet?
- `sameAs` does not contain `A`, so we continue
- `A` is added to `interfaceToNode`

## A`
- Next, the function is called on `A'`
- The SameAs interface `A` is found, and added to `sameAs`
- `A` is removed from `interfaceToNode` -- undoing what we did earlier?
- `sameAs` does contain `A`, so we return

## result
- `sameAs` contains both `A` and `A'`
- `interfaceToNode` contains neither ?

Would this function be easier if `interfaceToNode` was checked for the SameAs interface?  When called with `A`, the SameAs interface `A'` is not in `interfaceToNode`, and so `A'` is added.   When later called with `A'`, the SameAs interface `A` is already in `interfaceToNode`, and thus `A'` is not added?

Reply:
sameAs property only goes one way - from the interface we need (A) to the interface we don't need (A')

Reply:
oh!  OK.   now the code makes much more sense :)